storage/internal/queue.go

```diff
@@ -54,23 +63,24 @@ type FlushFunc func(ctx context.Context, entries []*tessera.Entry) error
 // the same order as they were added, when either the oldest entry in the queue has been there
 // for maxAge, or the size of the queue reaches maxSize.
 func NewQueue(ctx context.Context, maxAge time.Duration, maxSize uint, f FlushFunc) *Queue {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
You're shadowing the passed-in ctx here, which means the context-with-cancel is passed in to the worker thread on L107 rather than the outer context, is that ok?

Maybe rename to `cctx`?

Reply:
Yeah I think this is safer. Done.